File: problems/003-nth-fibonacci/rust/clean.rs
use crate::json::JVal;

pub fn solve(c: &JVal) -> JVal {
    let n = c.get("n").as_num() as i64;
    let (mut a, mut b) = (0i64, 1i64);
    for _ in 0..n { let t = a + b; a = b; b = t; }
    JVal::Num(a as f64)
}

File: problems/003-nth-fibonacci/c/clean.c
#define solve __glifex_ref_clean
#include "solution.h"

/* O(n) time, O(1) space: slide a two-element window (a, b) = (fib(k), fib(k+1))
   forward n times. The straightforward linear solution. */
JVal *solve(JVal *c) {
    long long n = (long long)jget(c, "n")->num;
    long long a = 0, b = 1;
    for (long long i = 0; i < n; i++) {
        long long t = a + b;
        a = b;
        b = t;
    }
    return jnum_((double)a);
}
